site/blog/2016/jul/1/sphinx-and-rtd-for-writers.rst

```diff
@@ -0,0 +1,552 @@
+* Referencing across pages, documents, and projects
+* Index and Glossary support
+* Internationalization support
+
+It also has practical and powerful tools for documenting software specifically,
+including:
+
+* Semantic referencing of software concepts, including classes, functions, programs, variables, etc.
+* Including code comments in documentation output for many programming languages
+* Tools for documenting HTTP APIs
+* Extensions with the Python language
+* A vast array of third party extensions to do useful things
+
+This article isn't large enough to cover all of the power packed into this tool.
+But I hope to show enough to pique reader interest so that you can try these tools out and research their capacity for yourself.
+
+Using Sphinx
+~~~~~~~~~~~~
+
+reStructuredText is a powerful language primarily because the syntax can be extended.
+reStructuredTest supports two types of extension:
+
+* Paragraph level (with Directives)
+* Inline level (with Interpreted Text Roles, or roles for short)
+
+Paragraph Level Markup
+``````````````````````
+
+Let's see a basic example of a *Directive* in reStructuredText::
+
+    .. warning:: Here be dragons! This topic covers a number of options that
+       might alter your database.
+
+       Proceed with caution!
+
+    .. figure:: screenshot-control-panel.jpg
+       :width: 50%
+
+       An overview of the admin control panel.
+
+Here ``warning`` acts as the name of the directive,
+and is the part you can extend for custom directives.
+In the ``figure``,
+``screenshot-control-panel.jpg`` is an *argument*,
+``:width:`` is an *option*,
+and the rest is the *content*
+They allow for customization of directives,
+and show the full power of reStructuredText.
+
+.. note:: You'll notice that Sphinx uses whitespace to denote where a directive ends.
+   The "Proceed with caution!" is still part of the ``warning``,
+   and everything that continues to be indented will be part of that warning.
+
+Sphinx ships with a number of powerful directives for documenting code.
+You can also write your own if you have someone on your team that knows Python.
+
+Inline Markup
+`````````````
+
+For extendability inside of a paragraph Sphinx uses roles.
+Here is an example:
+
+    You can learn more about this in :rfc:`1984`.
+    It is implemented in our code at :class:`System.Security`
+
+Here you can see ``rfc`` and ``class`` act as the name of the role,
+and then you can pass in a single argument.
+For the ``rfc`` role,
+it generates a link to the online reference for RFC 1984 with a text of ``RFC 1984``.
+
+The ``class`` is where things get interesting.
+This acts as a reference to the class ``System.Security`` that is documented in your project,
+which is a hyperlink in the HTML output,
+but also a working link in PDF and other outputs as well.
+
+You can implement your own roles with Python.
+This allows you to create powerful semantic constructs into the actual *markup* you're using to write documentation.
```

Comment:
`create ... into` sounds awkward. `create ... to add into`, `add ... into`, or somesuch.